src/core.ts

```diff
@@ -77,6 +84,11 @@ export default class Metrics {
     return ssoUrl
   }
 
+  async token (params: { clientIdentifier: string, clientSecret: string, authorizationCode?: string, refreshToken?: string, grantType: oAuthGrantTypes}) {
+    const response = await this._connection.action('oauth:token', { ...params }) as OAuthTokenResponse
+    return { accessToken: response.accessToken, refreshToken: response.refreshToken, expiresIn: response.expiresIn }
```

Comment:
Needs a typed result. Should probably result in a `UserSession` or `OAuthSession`.